File: app/src/main/java/com/saint/struct/network/HttpManager.java
package com.saint.struct.network;

import com.saint.struct.network.service.ConnectService;

public class HttpManager extends BaseHttpManager{
    private static final String URL="https://www.wanandroid.com/";
    private static final int TIMEOUT = 60;
    private final ConnectService mService;
    private HttpManager(int timeOut, String url) {
        super(timeOut, url);
        mService = getRetrofit().create(ConnectService.class);
    }

    //默认请求类
    public static HttpManager getInstance() {
        return SingletonHolder.INSTANCE;
    }

    private static class SingletonHolder {
        private static final HttpManager INSTANCE = new HttpManager(TIMEOUT,
                URL);
    }

    public ConnectService getService() {
        return mService;
    }
}

File: app/src/main/java/com/saint/struct/network/HttpManager.kt
package com.saint.struct.network

import com.saint.struct.network.service.ConnectService


class HttpManager private constructor(timeOut: Int, url: String) : BaseHttpManager(timeOut, url) {
    val service: ConnectService = retrofit.create(ConnectService::class.java)

    private object SingletonHolder {
        //默认请求类
        val instance = HttpManager(TIMEOUT, URL)
    }

    companion object {
        private const val URL = "https://www.wanandroid.com/"
        private const val TIMEOUT = 60
        @JvmStatic
        fun getInstance() = SingletonHolder.instance
    }
}